crates/core/src/schema/table_info.rs

```diff
@@ -0,0 +1,197 @@
+
+        let step_result = statement.step()?;
+        if step_result != ResultCode::ROW {
+            return Err(SQLiteError::from(ResultCode::SCHEMA));
+        }
+
+        let name = statement.column_text(0)?.to_string();
+        let view_name = statement.column_text(1)?.to_string();
+        let flags = {
+            let local_only = statement.column_int(2) != 0;
+            let insert_only = statement.column_int(3) != 0;
+            let include_metadata = statement.column_int(5) != 0;
+            let include_old_only_when_changed = statement.column_int(6) != 0;
+
+            let mut flags = TableInfoFlags::default();
+            flags = flags.set_flag(TableInfoFlags::LOCAL_ONLY, local_only);
+            flags = flags.set_flag(TableInfoFlags::INSERT_ONLY, insert_only);
+            flags = flags.set_flag(TableInfoFlags::INCLUDE_METADATA, include_metadata);
+            flags = flags.set_flag(
+                TableInfoFlags::INCLUDE_OLD_ONLY_IF_CHANGED,
+                include_old_only_when_changed,
+            );
+
+            flags
+        };
+
+        let include_old = match statement.column_type(4)? {
+            sqlite_nostd::ColumnType::Text => {
+                let columns: Vec<String> = serde_json::from_str(statement.column_text(4)?)?;
+                Some(DiffIncludeOld::OnlyForColumns { columns })
+            }
+
+            sqlite_nostd::ColumnType::Integer => {
+                if statement.column_int(4) != 0 {
+                    Some(DiffIncludeOld::ForAllColumns)
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        };
+
+        // Don't allow include_metadata for local_only tables, it breaks our trigger setup and makes
+        // no sense because these changes are never inserted into ps_crud.
+        if flags.include_metadata() && flags.local_only() {
+            return Err(SQLiteError(
+                ResultCode::ERROR,
+                Some("include_metadata and local_only are incompatible".to_string()),
+            ));
+        }
+
+        return Ok(TableInfo {
+            name,
+            view_name,
+            diff_include_old: include_old,
+            flags,
+        });
+    }
+}
+
+pub enum DiffIncludeOld {
+    OnlyForColumns { columns: Vec<String> },
+    ForAllColumns,
+}
+
+#[derive(Clone, Copy)]
+#[repr(transparent)]
+pub struct TableInfoFlags(u32);
+
+impl TableInfoFlags {
+    pub const LOCAL_ONLY: u32 = 1;
+    pub const INSERT_ONLY: u32 = 2;
+    pub const INCLUDE_METADATA: u32 = 4;
+    pub const INCLUDE_OLD_ONLY_IF_CHANGED: u32 = 8;
+
+    pub const fn local_only(self) -> bool {
+        self.0 & Self::LOCAL_ONLY != 0
+    }
+
+    pub const fn insert_only(self) -> bool {
+        self.0 & Self::INSERT_ONLY != 0
+    }
+
+    pub const fn include_metadata(self) -> bool {
+        self.0 & Self::INCLUDE_METADATA != 0
+    }
+
+    pub const fn include_old_only_if_changed(self) -> bool {
```

Comment:
Can we make this `include_old_only_when_changed` to make it consistent with the json option passed in?